pandas/core/index.py

```diff
@@ -2510,7 +2527,334 @@ def invalid_op(self, other=None):
 
 Index._add_numeric_methods_disabled()
 Index._add_logical_methods()
+Index._add_comparison_methods()
+
+class CategoricalIndex(Index):
+    """
+
+    Immutable Index implementing an ordered, sliceable set. CategoricalIndex
+    represents a sparsely populated Index with an underlying Categorical.
+
+    Parameters
+    ----------
+    data : array-like (1-dimensional)
+    categories : optional categories for the CategoricalIndex
+    copy : bool
+        Make a copy of input ndarray
+    name : object
+        Name to be stored in the index
+
+    """
+
+    _typ = 'categoricalindex'
+    _engine_type = _index.Int64Engine
+    _attributes = ['name','categories']
+
+    def __new__(cls, data=None, categories=None, dtype=None, copy=False, name=None, fastpath=False, ordered=None, **kwargs):
 
+        def create_categorical(data=data, categories=categories):
+            if categories is not None:
+                data = data.set_categories(categories)
+            if not data.ordered:
+                data = data.as_ordered()
+            return data
+
+        if fastpath:
+            return cls._simple_new(data, name=name)
+
+        if ordered is not None:
+            raise ValueError("CategoricalIndex are by definition ordered")
```

Comment:
its not, but ALL Index by definition are ordered. Just catching if someone passes ``ordered=False`` and expecting it to do something (it would simply be ignored on other Index creations, but it IS a valid keyword to Categoricals, so though the raise makes sense).